### PR TITLE
feat(Core/Commands): Gear

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
@BarbzYHOOL
OLD PR: https://github.com/IntelligentQuantum/azerothcore-wotlk/commit/fdb85ebf2d571dfbee9531449c0f8e569be17ac8

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4105

## Tests Performed:

- Tested in game
- Arch Linux

## How to Test the Changes:

1. .gear repair
2. .gear stats (PlayerSave.Stats.MinLevel = 0) Shows only average item level
3. If you want to see player stats enable this PlayerSave.Stats.MinLevel = 1

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Helias — APPROVED** [review]

File: src/server/scripts/Commands/cs_misc.cpp [hunk 1]
```diff
@@ -3385,6 +3390,47 @@ class misc_commandscript : public CommandScript
...
+
+    static bool HandleGearStatsCommand(ChatHandler* handler, char const* /*args*/)
+    {
+        Player* player = handler->getSelectedPlayerOrSelf();
+
+        if (!player)
+            return false;
+
+        handler->PSendSysMessage("Character: %s", player->GetPlayerName().c_str());
+        handler->PSendSysMessage("Current equipment average item level: |cff00ffff%u|r", (int)player->GetAverageItemLevel());
```
Comment (Kitzunu) [inline src/server/scripts/Commands/cs_misc.cpp:3402]:
```suggestion
        handler->PSendSysMessage("Current equipment average item level: |cff00ffff%u|r", (int32)player->GetAverageItemLevel());
```

File: src/server/scripts/Commands/cs_misc.cpp [hunk 2]
```diff
@@ -3385,6 +3390,47 @@ class misc_commandscript : public CommandScript
             return true;
         }
     }
+
+    static bool HandleGearStatsCommand(ChatHandler* handler, char const* /*args*/)
+    {
+        Player* player = handler->getSelectedPlayerOrSelf();
+
+        if (!player)
+            return false;
```
Comment (Kitzunu) [inline src/server/scripts/Commands/cs_misc.cpp:3399]: brackets

**Review by Kitzunu — CHANGES_REQUESTED** [review]

**Comment by FrancescoBorzi** [comment]
@IntelligentQuantum I guess no need for further testing as you have already tested it in game and it doesn't seem big change. What do you think?

**Comment by IntelligentQuantum** [comment]
@FrancescoBorzi We can merge it, if you have not problem with .gear stats because player can use this command

**Comment by Kitzunu** [comment]
Please update the wiki as well

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**